### PR TITLE
Add `pack set-default-registry` command

### LINKED ISSUE #745: Buildpack Registry config commands
### Description

As a `pack` user it would be nice to have commands to manage the registries in the `config` file.

### Proposed solution

- [x] `pack list-buildpack-registry` - a new command that will list all the `[[registries]]` config in `.pack/config.toml`.
- [x] `pack add-buildpack-registry` - a new command that will add to the `[[registries]]` config in `.pack/config.toml`.
- [ ] `pack remove-buildpack-registry` - a new command that will remove a registry from the `[[registries]]` config in `.pack/config.toml`.
- [ ] `pack set-buildpack-registry` - a new command that will mutate the `default-registry` config in `.pack/config.toml`.

For more details on what the data structure of the `.pack/config.toml` see [RFC #44](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#pack-register-buildpack-url).

### Additional Refactor/Clean-up
`add-buildpack-registry `-> `add-registry`
`list-buildpack-registry` -> `list-registries`

### Additional context
- [x] This feature should be documented somewhere

- These commands should be hidden via `experimental` config option.

### PR DESCRIPTION
**Pull request by elbandito**
Signed-off-by: Travis <longoria.public@gmail.com>

## Summary
This PR introduces a new `pack` command to allow the user to set the default registry name via `pack set-registry`.

#### New

```
❯ pack
CLI for building apps using Cloud Native Buildpacks

Usage:
  pack [command]

Available Commands:
  set-default-registry              Set default registry
```

```
❯ pack set-default-registry --help
Set default registry

Usage:
  pack set-default-registry <name> [flags]

Examples:
pack set-default-registry myregistry

Flags:
  -h, --help   Help for 'set-registry'

Global Flags:
      --no-color     Disable color output
  -q, --quiet        Show less output
      --timestamps   Enable timestamps in output
  -v, --verbose      Show more output
```

## Documentation

- Should this change be documented?
    - [ X ] Yes, see #___
    - [ ] No

## Related

Resolves #745

### CODE REVIEW

File: internal/commands/set_registry.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	)
+
+	cmd := &cobra.Command{
+		Use:   "set-registry <name>",
+		Args:  cobra.ExactArgs(1),
+		Short: "Set default registry",
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			registryName = args[0]
+			if !containsRegistry(config.GetRegistries(cfg), registryName) {
+				return errors.Newf("no registry with the name %s exists", style.Symbol(registryName))
```
Comment (jromero) [inline internal/commands/set_registry.go:24]: I don't think we should be importing yet another error lib for a single function.

`pkg/errors` provides this functionality via `errors.Errorf`.

File: internal/commands/set_registry.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/buildpacks/pack/logging"
+)
+
+func SetRegistry(logger logging.Logger, cfg config.Config, cfgPath string) *cobra.Command {
+	var (
+		registryName string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "set-registry <name>",
```
Comment (jromero) [inline internal/commands/set_registry.go:18]: Based on already having `set-default-builder`, this should be `set-default-registry` for symmetry.

**Review by jromero — CHANGES_REQUESTED** [review]

File: internal/commands/set_registry.go (hunk 1 above)
Comment (elbandito) [reply internal/commands/set_registry.go:24]: Agreed.  I must have accidentally done this.

File: internal/commands/set_registry.go (hunk 2 above)
Comment (elbandito) [reply internal/commands/set_registry.go]: I like that

File: internal/commands/set_default_registry_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			_ = os.RemoveAll(tmpDir)
+		})
+
+		it("should set the default registry", func() {
+			cfg := config.Config{
+				Registries: []config.Registry{
+					{
+						Name: "myregistry",
+						URL:  "https://github.com/buildpacks/registry-index",
+						Type: "Github",
```
Comment (dfreilich) [inline internal/commands/set_default_registry_test.go:55]: Based on [this](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#type-github-schema), it should be `github`

```suggestion
						Type: "github",
```

**Review by dfreilich — CHANGES_REQUESTED** [review]
2 minor nits, but overall looks great!

**Comment by dfreilich** [comment]
One more nit: can we move the `cmd.Example` to be part of the definition of the cmd?

**Comment by elbandito** [comment]
@jromero and @dfreilich I made changes per comments.  Can I get a final look and blessing?

**Review by jromero — APPROVED** [review]

**Pull request merged**